app/mailers/patient_mailer.rb

```diff
@@ -169,18 +177,23 @@ def assessment_email(patient)
       return
     end
 
+    # Cover potential race condition where multiple messages are sent for the same monitoree.
+    return unless patient.last_assessment_reminder_sent_eligible?
+
     @lang = patient.select_language
     # Gather patients and jurisdictions
     # patient.dependents includes the patient themselves if patient.id = patient.responder_id (which should be the case)
     @patients = patient.active_dependents.uniq.collect do |dependent|
       { patient: dependent, jurisdiction_unique_id: Jurisdiction.find_by_id(dependent.jurisdiction_id).unique_identifier }
     end
+
+    update_last_assessment_reminder_sent(patient) # Update last send attempt timestamp before SMTP call
     mail(to: patient.email&.strip, subject: I18n.t('assessments.email.reminder.subject', locale: @lang || :en)) do |format|
       format.html { render layout: 'main_mailer' }
     end
     add_success_history(patient, patient)
-    # Always update the last contact time so the system does not try and send emails again.
-    patient.update(last_assessment_reminder_sent: DateTime.now)
+  rescue StandardError
+    clear_last_assessment_reminder_sent(patient) # Reset send attempt timestamp on failure
```

Comment:
Since this is new functionality in that we are rescuing all possible errors in this case, should we continue to `raise` after we clear?

Reply:
Agreed, made this change.